starlette/applications.py

```diff
@@ -108,20 +108,13 @@ def build_middleware_stack(self) -> ASGIApp:
     def routes(self) -> typing.List[BaseRoute]:
         return self.router.routes
 
-    @property
-    def debug(self) -> bool:
-        return self._debug
-
-    @debug.setter
-    def debug(self, value: bool) -> None:
-        self._debug = value
-        self.middleware_stack = self.build_middleware_stack()
-
     def url_path_for(self, name: str, **path_params: typing.Any) -> URLPath:
         return self.router.url_path_for(name, **path_params)
 
     async def __call__(self, scope: Scope, receive: Receive, send: Send) -> None:
         scope["app"] = self
+        if self.middleware_stack is None:
+            self.middleware_stack = self.build_middleware_stack()
```

Comment:
```suggestion
        if scope["type"] != "lifespan" and self.middleware_stack is None:
            self.middleware_stack = self.build_middleware_stack()
```

Reply:
What if the middleware runs something on lifespan? Maybe we need to build once before running the lifespan and then, if it's changed, we re-build it after running the lifespan?

Reply:
Actually we want both scenarios...

Reply:
What do you mean?

Reply:
Yeah :P

Reply:
> What do you mean?

The same thing as your first message.

But actually... We don't want to rebuild, otherwise we introduce the issue that motivated this PR again.